### PR TITLE
Add a citus_is_coordinator function to check whether a node is the coordinator

### PR DESCRIPTION
**Pull request by marcocitus**
DESCRIPTION: Add a citus_is_coordinator function to check whether a node is the coordinator

It can be useful for an application to know whether it is connected to the coordinator, such that it can redirect certain unsupported operations.

### CODE REVIEW

File: src/backend/distributed/sql/citus--11.0-1--11.0-2.sql [hunk 1]
```diff
@@ -1 +1 @@
--- bump version to 11.0-2
+#include "udfs/citus_is_coordinator/11.0-2.sql"
```
Comment (tejeswarm) [inline src/backend/distributed/sql/citus--11.0-1--11.0-2.sql:1]: The current version is '11.1-1', so this should be in citus--11.0-2--11.1-1.sql ?

Comment (tejeswarm) [reply src/backend/distributed/sql/citus--11.0-1--11.0-2.sql:1]: You are dropping from citus--11.1-1--11.0-2.sql

File: src/backend/distributed/metadata/node_metadata.c [hunk 2]
```diff
@@ -150,6 +150,7 @@ PG_FUNCTION_INFO_V1(get_shard_id_for_distribution_column);
 PG_FUNCTION_INFO_V1(citus_nodename_for_nodeid);
 PG_FUNCTION_INFO_V1(citus_nodeport_for_nodeid);
 PG_FUNCTION_INFO_V1(citus_coordinator_nodeid);
+PG_FUNCTION_INFO_V1(citus_is_coordinator);
```
Comment (onderkalaci) [inline src/backend/distributed/metadata/node_metadata.c:153]: Should we maybe introduce `get_current_host_info` or such which returns the the tuple (hostname, hostport, nodeid, is_coordinator)?
(we can keep this as well, and we can do it with a separate pr, mostly asking your thoughts)

Comment (onderkalaci) [reply src/backend/distributed/metadata/node_metadata.c:153]: Well, never mind, we can achieve this with simple SQL queries anyway:
```SQL
SELECT *, 
	CASE WHEN groupid = (SELECT groupid FROM pg_dist_local_group) THEN '*' 
		 ELSE ' ' END current_node 
FROM pg_dist_node;

┌───────────┬──────────┬──────────────┐
│ nodename  │ nodeport │ current_node │
├───────────┼──────────┼──────────────┤
│ localhost │     9700 │              │
│ localhost │     9701 │ *            │
│ localhost │     9702 │              │
│ localhost │     5432 │              │
└───────────┴──────────┴──────────────┘
(4 rows)
```

File: src/backend/distributed/sql/citus--11.0-1--11.0-2.sql (hunk 1 above)
Comment (marcocitus) [reply src/backend/distributed/sql/citus--11.0-1--11.0-2.sql:1]: We will backport this to release-11.0 which is currently at 11.0-1 (for the beta).

**Comment by marcocitus** [comment]
> Is that a caveat you are describing where a non-coordinator can have group ID 0?

yes, a worker node in a cluster in which metadata syncing is disabled, the pg_dist_local_group would be set to 0.

**Comment by marcocitus** [comment]
> Should we return DATUM?

the PG_RETURN_ macros will convert to Datum

**Review by tejeswarm — APPROVED** [review]
